dataprofiler/reports/graphs.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+import math
+import warnings
+
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+
+
+def plot_histograms(profiler, columns=None):
+    """
+    Take a input of StructuredProfiler class and a list of specified column
+    names and then plots the histograms of those that are int or float
+    columns.
+
+    :param profiler: StructuredProfiler variable
+    :param columns: list of column names to be plotted
+    :type profiler: StructuredProfiler
+    :type columns: list
+    :return: returns fig
+    :rtype: fig
+    """
+
+    # get all inds to graph, raise error if user specified doesn't exist (part 1)
+    inds_to_graph = []
+    if not columns:
+        inds_to_graph = list(range(len(profiler.profile)))
+    else:
+        for column in columns:
+            col = column
+            if isinstance(col, str):
+                col = col.lower()
+            if col not in profiler._col_name_to_idx:
+                raise ValueError("Column \"" + str(col) + "\" is not found as a "
+                                                       "profiler column")
+            inds_to_graph.extend(profiler._col_name_to_idx[col])
+        sorted(inds_to_graph)
+    # get all columns which are either int or float (part2)
+    for col_ind in reversed(inds_to_graph):
+        # get data_type
+        data_compiler = profiler.profile[col_ind].profiles['data_type_profile']
+        data_type = data_compiler.selected_data_type
+        data_type_profiler = data_compiler._profiles[data_type]
+        if data_type not in ['int', 'float']:
+            inds_to_graph.pop()
```

Comment:
Have you considered using `filter` on a list, to just hand it a boolean function and it filters out only values that return True. I think that could clean up this section a good bit.

Reply:
Also it seems `data_type_profiler` never gets used, you could probably remove that

Reply:
And I'm not sure this is working as intended, because you iterate backwards, and pop if something isn't int/float, but that wont work because if you get to an item that is int/float, then the next isn't, you'll actually pop the one that is.

Reply:
Regardless of how you solve it though, should add a test case where some of the users specified columns aren't int/float, and verify that they get properly ignored, and nothing that is actually int/float gets omitted on accident.

Reply:
```python
    def is_index_graphable_column(ind_to_graph):
        col_profiler = profiler.profile[ind_to_graph]
        data_compiler = col_profiler.profiles['data_type_profile']
        if data_compiler.selected_data_type not in ['int', 'float']:
            return False
        return True
    inds_to_graph = list(filter(is_index_graphable_column, inds_to_graph))
```
so like this?